airbyte-cdk/python/airbyte_cdk/sources/declarative/parsers/manifest_component_transformer.py

```diff
@@ -89,60 +88,47 @@ def propagate_types_and_parameters(
-            # have no low-code connectors that use class_name except for custom components.
             if "class_name" in propagated_component:
                 found_type = CUSTOM_COMPONENTS_MAPPING.get(parent_field_identifier)
             else:
                 found_type = DEFAULT_MODEL_TYPES.get(parent_field_identifier)
             if found_type:
                 propagated_component["type"] = found_type
 
-        # When there is no resolved type, we're not processing a component (likely a regular object) and don't need to propagate parameters
-        # When the type refers to a json schema, we're not processing a component as well. This check is currently imperfect as there could
-        # be json_schema are not objects but we believe this is not likely in our case because:
-        # * records are Mapping so objects hence SchemaLoader root should be an object
-        # * connection_specification is a Mapping
         if "type" not in propagated_component or self._is_json_schema_object(propagated_component):
             return propagated_component
 
-        # Combines parameters defined at the current level with parameters from parent components. Parameters at the current
-        # level take precedence
-        current_parameters = dict(copy.deepcopy(parent_parameters))
+        current_parameters = parent_parameters.copy()
         component_parameters = propagated_component.pop(PARAMETERS_STR, {})
-        current_parameters = {**current_parameters, **component_parameters}
+        current_parameters.update(component_parameters)
 
-        # Parameters should be applied to the current component fields with the existing field taking precedence over parameters if
-        # both exist
         for parameter_key, parameter_value in current_parameters.items():
-            propagated_component[parameter_key] = propagated_component.get(parameter_key) or parameter_value
+            propagated_component.setdefault(parameter_key, parameter_value)
 
         for field_name, field_value in propagated_component.items():
             if isinstance(field_value, dict):
-                # We exclude propagating a parameter that matches the current field name because that would result in an infinite cycle
                 excluded_parameter = current_parameters.pop(field_name, None)
                 parent_type_field_identifier = f"{propagated_component.get('type')}.{field_name}"
                 propagated_component[field_name] = self.propagate_types_and_parameters(
                     parent_type_field_identifier, field_value, current_parameters
                 )
                 if excluded_parameter:
                     current_parameters[field_name] = excluded_parameter
-            elif isinstance(field_value, typing.List):
-                # We exclude propagating a parameter that matches the current field name because that would result in an infinite cycle
+            elif isinstance(field_value, list):
```

Comment:
```suggestion
            elif isinstance(field_value, list):
                # We exclude propagating a parameter that matches the current field name because that would result in an infinite cycle
```